controllers/managedocs_controller.go

```diff
@@ -361,9 +361,7 @@ func (r *ManagedOCSReconciler) initReconciler(ctx context.Context, req ctrl.Requ
 	r.alertmanager.Name = alertmanagerName
 	r.alertmanager.Namespace = r.namespace
 
-	r.addonParamSecret = &corev1.Secret{}
-	r.addonParamSecret.Name = r.AddonParamSecretName
-	r.addonParamSecret.Namespace = r.namespace
+	r.addonParams = make(map[string]string)
```

Comment:
Can we move this to the end of the init functions (or the start)? right now it is in the middle of resource initializations and it breaks the reading flow.